src/main/java/seedu/storemando/model/item/ItemExpiringPredicate.java

```diff
@@ -0,0 +1,36 @@
+package seedu.storemando.model.item;
+
+import static java.time.temporal.ChronoUnit.DAYS;
+
+import java.time.LocalDate;
+import java.util.function.Predicate;
+
+/**
+ * Tests that a {@code Item}'s {@code expiry date} is within the stipulated days from today.
+ */
+public class ItemExpiringPredicate implements Predicate<Item> {
+    private final Long numOfDays;
+
+    public ItemExpiringPredicate(Long numOfDays) {
+        this.numOfDays = numOfDays;
+    }
+
+    @Override
+    public boolean test(Item item) {
+        try {
+            LocalDate itemExpiryDate = item.getExpiryDate().expiryDate;
```

Comment:
Does this violate the law of demeter since you are getting an object from item and accessing its field?

Reply:
So I should break into 2 statements?
ExpiryDate expiryDate = item.getExpiryDate();
LocalDate itemExpiryDate = expiryDate.expiryDate;

Reply:
I believe Item should provide a method that retrieves expiryDate directly.